src/OSPSuite.Core/Domain/Services/ParameterIdentifications/CategorialParameterIdentificationRunInitializer.cs

```diff
@@ -28,24 +37,33 @@ public void Initialize(ParameterIdentification parameterIdentification, int runI
          _runMode = parameterIdentification.Configuration.RunMode.DowncastTo<CategorialParameterIdentificationRunMode>();
       }
 
-      public override Task<ParameterIdentification> InitializeRun()
+      public override Task<ParameterIdentification> InitializeRun(CancellationToken cancellationToken)
       {
+         var parallelOptions = new ParallelOptions
+         {
+            CancellationToken = cancellationToken,
+            MaxDegreeOfParallelism = _coreUserSettings.MaximumNumberOfCoresToUse
+         };
+
          return Task.Run(() =>
          {
             var newParameterIdentification = _cloneManager.Clone(ParameterIdentification);
-            newParameterIdentification.AllSimulations.ToList().Each(originalSimulation =>
+            Parallel.ForEach(newParameterIdentification.AllSimulations.ToList(), parallelOptions, originalSimulation =>
```

Comment:
Simulations are now cloned in parallel instead of one at a time. This is a very expensive operation.